src/Routing/Route/DashedRoute.php

```diff
@@ -49,31 +50,47 @@ public function parse($url)
         if (!$params) {
             return false;
         }
-        if (!empty($params['controller'])) {
-            $params['controller'] = Inflector::camelize(str_replace(
-                '-',
-                '_',
```

Comment:
I think... a much lower cost version of what you've done here is to change this to:

    if (!empty($params['controller'])) {
        if (strpos($params['controller'], '_') !== false ||
            strtolower($params['controller']) !== $params['controller']
        ) {
            return false;
        }
        $params['controller'] = Inflector::camelize(str_replace('-', '_', $params['controller']));
    }

I.e. if you want the route to be stricter - just bail-early for anything that looks to be wrong.